tools/dotnet-linker/ApplyPreserveAttributeBase.cs

```diff
@@ -259,9 +270,11 @@ void AddDynamicDependencyAttribute (TypeDefinition type, bool allMembers)
 			abr.SetCurrentAssembly (type.Module.Assembly);
 
 			var moduleConstructor = GetOrCreateModuleConstructor (type.GetModule ());
-			var members = allMembers
-				? DynamicallyAccessedMemberTypes.All
-				: DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors;
+			var members = (type, allMembers) switch {
+				( { IsEnum: true }, _) => DynamicallyAccessedMemberTypes.PublicFields,
+				(_, false) => DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors,
+				(_, true) => allMemberTypes,
+			};
```

Comment:
nit: I find the pattern matching switch kinda hard to read compared to splitting it up a bit:

```csharp
var members = allMembers
	? allMemberTypes
	: DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors;

// only preserve fields for enums
if (type.IsEnum)
	members = DynamicallyAccessedMemberTypes.PublicFields
```